src/designer/designer_packages/elsa-workflows-designer/src/components/shared/form-panel/form-panel.tsx

```diff
@@ -2,11 +2,18 @@ import {Component, Event, EventEmitter, h, Prop} from '@stencil/core';
 import {TabChangedArgs, TabDefinition} from '../../../models';
 import {isNullOrWhitespace} from "../../../utils";
 import {PanelActionClickArgs, PanelActionDefinition, PanelActionType} from "./models";
+import {WorkflowDefinitionsApi} from '../../../modules/workflow-definitions/services/api';
+import Container from 'typedi';
+import studioComponentStore from '../../../data/studio-component-store';
+import {WorkflowDefinition} from '../../../interfaces';
+import workflowStore from '../../../data/workflow-store';
 
 @Component({
   tag: 'elsa-form-panel'
 })
 export class FormPanel {
+  private workflowDefinitionsApi: WorkflowDefinitionsApi;
```

Comment:
The `FormPanel` is a reusable component that should be agnostic to workflows, and remain simple containers.
Can you move the stuff you added to the component that is contained in the form panel? Probably in the `elsa-workflow-definition-editor` component?

Reply:
Got it, will do the change 👍🏻

Reply:
Just wanted to ask if it is better to move the logic to 'elsa-workflow-definition-properties-editor' component instead of 'elsa-workflow-definition-editor' to avoid prop drilling?

Reply:
Maybe, but I wonder if it makes sense for this logic to be part of the `elsa-workflow-definition-properties-editor` - does that component have anything to do with navigation between workflow definition levels?

Reply:
Nope, you are right nothing related to properties tab of the workflow.